source/addon.cpp

```diff
@@ -68,11 +68,48 @@ bool ReShadeGetConfigValue(HMODULE, reshade::api::effect_runtime *runtime, const
 	else if (*size != 0)
 	{
 		*size = value_string.copy(value, *size - 1);
-		value[*size++] = '\0';
+		value[*size] = '\0';
 	}
 
 	return true;
 }
+bool ReShadeGetConfigValue2(HMODULE, reshade::api::effect_runtime *runtime, const char *section, const char *key, char *value, size_t *size)
```

Comment:
Returning the value list as a buffer of consecutive of null-terminated items is a nice solution, it would actually allow making this a drop-in change without having to introduce a new export! Since previously `ReShadeGetConfigValue` already returned the buffer size including the null-terminator, expanding it to be able to return a buffer with multiple null-terminators would not be a problem, nor would it break existing add-ons, since those would only operate on settings with scalar values (and therefore behavior would stay the same).
So easy to move the new implementation of `ReShadeGetConfigValue2` into `ReShadeGetConfigValue` and get rid of the added `ReShadeGetConfigValue2`.

Reply:
You can understand what I mean as an example how to write the list in the Win32 API's PrivateProfile functions and ImGui.
Arrays end with '\0' , and strings end with '\0', so if you expand to handle multiple elements, the terminator will be "\0\0", even if contains a single element.
This would change the specification of the size advisement, but maybe it is a good time to change it now, depending the how actual users of `get/set_config_value`?

Reply:
I had been coming from the assumption that the null-terminator was already included in the returned size, in which case having an additional one in there wouldn't really be a problem. Since that isn't actually the case, this does change behavior. Though looking through the list of add-ons the setup tool advertises, it doesn't look like any of them would break from such a change, so probably still ok. The other overloads of `get/set_config_value` use `std::from_chars`, which I believe can handle a null character at the end.